servers/rendering/renderer_rd/environment/gi.cpp

```diff
@@ -1928,8 +1928,8 @@ void GI::SDFGI::pre_process_gi(const Transform3D &p_transform, RenderDataRD *p_r
 		}
 
 		AABB cascade_aabb;
-		cascade_aabb.position = Vector3((Vector3i(1, 1, 1) * -int32_t(cascade_size >> 1) + cascade.position)) * cascade.cell_size;
-		cascade_aabb.size = Vector3(1, 1, 1) * cascade_size * cascade.cell_size;
+		cascade_aabb.position = Vector3((Vector3i(1, 1, 1) * -int32_t(cascade_size >> 1) + cascade.position)) * cascade.cell_size * Vector3(1, 1.0 / y_mult, 1);
+		cascade_aabb.size = Vector3(1, 1, 1) * cascade_size * cascade.cell_size * Vector3(1, 1.0 / y_mult, 1);
```

Comment:
Small optimization (the multiplier doesn't change across cascades, so this could make the logic easier to understand too):

```suggestion
		cascade_aabb.position = Vector3((Vector3i(1, 1, 1) * -int32_t(cascade_size >> 1) + cascade.position)) * cascade.cell_size * y_scale;
		cascade_aabb.size = Vector3(1, 1, 1) * cascade_size * cascade.cell_size * y_scale;
```

Add this before the `for` loop:

```cpp
const Vector3 y_mult = Vector3(1, 1.0 / y_mult, 1);
```

I've tested this change locally and it works the same.

Reply:
To clarify, is this what you mean?

```diff
diff --git a/servers/rendering/renderer_rd/environment/gi.cpp b/servers/rendering/renderer_rd/environment/gi.cpp
index 6944c79962..697cc689c0 100644
--- a/servers/rendering/renderer_rd/environment/gi.cpp
+++ b/servers/rendering/renderer_rd/environment/gi.cpp
@@ -1928,8 +1928,11 @@ void GI::SDFGI::pre_process_gi(const Transform3D &p_transform, RenderDataRD *p_r
                }

                AABB cascade_aabb;
-               cascade_aabb.position = Vector3((Vector3i(1, 1, 1) * -int32_t(cascade_size >> 1) + cascade.position)) * cascade.cell_size * Vector3(1, 1.0 / y_mult, 1);
-               cascade_aabb.size = Vector3(1, 1, 1) * cascade_size * cascade.cell_size * Vector3(1, 1.0 / y_mult, 1);
+
+               const Vector3 y_scale = Vector3(1, 1.0 / y_mult, 1);
+
+               cascade_aabb.position = Vector3((Vector3i(1, 1, 1) * -int32_t(cascade_size >> 1) + cascade.position)) * cascade.cell_size * y_scale;
+               cascade_aabb.size = Vector3(1, 1, 1) * cascade_size * cascade.cell_size * y_scale;

                for (uint32_t j = 0; j < p_render_data->sdfgi_update_data->positional_light_count; j++) {
                        if (idx == SDFGI::MAX_DYNAMIC_LIGHTS) {
```

Reply:
No, I mean putting the `y_scale` assignment outside the `for` loop it's currently in. In your PR, this would be on line 1883.